src/NServiceBus.Core/Logging/ColoredConsoleLogger.cs

```diff
@@ -1,26 +1,26 @@
 ﻿namespace NServiceBus
 {
     using System;
-    using System.Runtime.InteropServices;
+    using System.IO;
     using Logging;
 
     static class ColoredConsoleLogger
     {
         static ColoredConsoleLogger()
         {
-            const int STD_OUTPUT_HANDLE = -11;
-            logToConsole = GetStdHandle(STD_OUTPUT_HANDLE) != IntPtr.Zero;
+            using (var stream = Console.OpenStandardOutput())
+            {
+                logToConsole = stream != Stream.Null;
```

Comment:
if you run a WPF app in debug mode, it seems like there is still a standard output stream available. When no debugger is attached, this is false as "intended" (while userinteractive is still true) so I guess this is closest to what we want.